doc/api_test_driver.rst

```diff
@@ -309,3 +320,143 @@ test fragments.
+  another one, the testsuite needs to make sure that this other test is also
+  run.
+
+Most of the time, these drawbacks make inter-test dependencies inappropriate,
+and thus better avoided. However there are cases where they are necessary. Real
+world examples include:
+
+* Writing an ``e3-testsuite`` based test harness to exercize existing
+  inter-dependent testcases that cannot be modified. For instance, the `ACATS
+  (Ada Conformity Assessment Test Suite) <http://www.ada-auth.org/acats.html>`_
+  has some tests which write files and other tests that then read later on.
+
+* External constraints require separate tests to host the validation of data
+  produced in other tests. For instance a qualification testsuite (in the
+  context of software certification) that needs a single test (say
+  ``report-format-check``) to check that all the outputs of a qualified tool
+  throughout the testsuite (say output of tests ``feature-A``, ``feature-B``,
+  ...) respect a given constraint.
+
+  Notice how, in this case, the outcome of such a test depends on how the
+  testsuite is run: if ``report-format-check`` detects a problem in the output
+  from ``feature-A`` but not in outputs from other tests, then
+  ``report-format-check`` will pass or fail depending on the specific set of
+  tests that the testsuite is requested to run.
+
+With these pitfalls in mind, let's see how to create inter-test dependencies.
+First, a bit of theory regarding the logistics of test fragments in the
+testsuite:
+
+The description of the :ref:`TestDriver.add_fragment method
+<test_driver_add_fragment>` above mentionned a crucial data structure in the
+testsuite: the DAG (Directed Acyclic Graph). This graph (an instance of
+``e3.collections.dag.DAG``) contains the list of fragments to run as nodes and
+the dependencies between these fragments as edges. The DAG is then is used to
+schedule their execution: first execute fragments that have no dependencies,
+then fragments that depend on these, etc.
+
+Each node in this graph is a ``FragmentData`` instance, that the
+``add_fragment`` method creates. This class has four fields:
+
+* ``uid``, a string used as an identifier for this fragment that is unique in
+  the whole DAG (it corresponds to the ``VertexID`` generic type in
+  ``e3.collections.dag``). ``add_fragment`` automatically creates it from the
+  driver's ``test_name`` field and ``add_fragment``'s own ``name`` argument.
+
+* ``driver``, the test driver that created this fragment.
+
+* ``name``, the ``name`` argument passed to ``add_fragment``.
+
+* ``callback``, the ``fun`` argument passed to ``add_fragment``.
+
+Our goal here is, once the DAG is populated with all the ``FragmentData`` to
+run, to add dependencies between them to express scheduling constraints.
+Overriding the ``Testsuite.adjust_dag_dependencies`` method allows this: this
+method is called the DAG was created and populated, and right before the
```

Comment:
Missing `when`?

Reply:
Ah yes, thanks! And thank you for letting me know that you have an idea of how to use this new API. :-)